### PR TITLE
move to token definition struct containing type/name/is-a bools

### PR DESCRIPTION
**Pull request by joram**

### CODE REVIEW

File: jinja/parser.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package jinja
+
+type Parser struct {
+	tokens    chan Token
+	headToken Token
+	tokenizer Tokenizer
+	nodes     []NodeInter
+}
+
+func (parser *Parser) Parse(template string) {
```
Comment (dlsteuer) [inline jinja/parser.go:10]: This should take a tokenizer instead of a string i.e. `Parse(t Tokenizer)`

File: jinja/node.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+func (n VariableNode) isComplete() bool { return true }
+
+func (n VariableNode) String() string { return "<VAR: " + n.variableName + ">" }
+
+type ValueNode struct {
+	value interface{}
+}
+
+func (n ValueNode) isComplete() bool { return true }
```
Comment (dlsteuer) [inline jinja/node.go:48]: I think the term Literal is more explicit over Value

File: jinja/node.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+func (n Node) String() string {
+	s := ""
+	for _, t := range n.tokens {
+		s += fmt.Sprintf("'%+v'", t.String())
+	}
+	return s
+}
+
+type ExpressionNode struct {
```
Comment (dlsteuer) [inline jinja/node.go:26]: This is technically a BinaryNode because it's for a binary operator with a left and right child.

File: jinja/node.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package jinja
+
+import "fmt"
+
+type NodeInter interface {
+	String() string
+}
+
+type Node struct {
```
Comment (dlsteuer) [inline jinja/node.go:9]: Node should probably be an interface, and remove NodeInter, and then specific node types can implement the Node interface.

File: jinja/parser.go (hunk 1 above)
Comment (joram) [reply jinja/parser.go:10]: +1

**Pull request closed without merging**